x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -3243,23 +3243,21 @@ public void testPushWhereEvalToSource() {
             """;
 
         var plan = this.physicalPlan(query, airports);
-        var limit = as(plan, LimitExec.class);
+        var eval = as(plan, EvalExec.class);
+        var limit = as(eval.child(), LimitExec.class);
         var exchange = as(limit.child(), ExchangeExec.class);
         var fragment = as(exchange.child(), FragmentExec.class);
         var limit2 = as(fragment.fragment(), Limit.class);
         var filter = as(limit2.child(), Filter.class);
         assertThat("filter contains LessThan", filter.condition(), instanceOf(LessThan.class));
 
         var optimized = optimizedPlan(plan);
-        var topLimit = as(optimized, LimitExec.class);
+        eval = as(optimized, EvalExec.class);
+        var topLimit = as(eval.child(), LimitExec.class);
         exchange = as(topLimit.child(), ExchangeExec.class);
         var project = as(exchange.child(), ProjectExec.class);
         var fieldExtract = as(project.child(), FieldExtractExec.class);
         assertThat(fieldExtract.attributesToExtract().size(), greaterThan(5));
-        limit = as(fieldExtract.child(), LimitExec.class);
-        var eval = as(limit.child(), EvalExec.class);
-        fieldExtract = as(eval.child(), FieldExtractExec.class);
-        assertThat(fieldExtract.attributesToExtract().stream().map(Attribute::name).collect(Collectors.toList()), contains("scalerank"));
         var source = source(fieldExtract.child());
         var condition = as(source.query(), SingleValueQuery.Builder.class);
```

Comment:
nit: We could assert that the limit is now in the EsQueryExec for completeness in both tests.